sdk/kiosk/src/client/kiosk-client.ts

```diff
@@ -0,0 +1,485 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type SuiClient } from '@mysten/sui.js/src/client';
+import { fetchKiosk, getOwnedKiosks } from '../query/kiosk';
+import {
+	type FetchKioskOptions,
+	KioskData,
+	KioskOwnerCap,
+	ObjectArgument,
+	OwnedKiosks,
+	Network,
+	type KioskClientOptions,
+} from '../types';
+import { TransactionArgument, TransactionBlock } from '@mysten/sui.js/transactions';
+import * as kioskTx from '../tx/kiosk';
+import { TransferPolicyRule, personalKioskAddress, rules } from '../constants';
+import { queryTransferPolicy } from '../query/transfer-policy';
+import { convertToPersonalTx } from '../tx/personal-kiosk';
+import { confirmRequest } from '../tx/transfer-policy';
+import { objArg } from '../utils';
+
+export type PurchaseOptions = {
+	extraArgs?: Record<string, any>;
+};
+
+/**
+ * A Client that allows you to interact with kiosk.
+ * Offers utilities to query kiosk, craft transactions to edit your own kiosk,
+ * purchase, manage transfer policies, create new kiosks etc.
+ */
+export class KioskClient {
+	client: SuiClient;
+	network: Network;
+	#rules: TransferPolicyRule[];
+	selectedCap?: KioskOwnerCap;
+
+	constructor(options: KioskClientOptions) {
+		this.client = options.client;
+		this.network = options.network;
+		this.#rules = rules; // add all the default rules. WIP
+	}
+
+	/**
+	 * Should be called before running any transactions that involve kiosk.
+	 * @param cap The cap object, as returned from `getOwnedKiosks`.
+	 */
+	setSelectedCap(cap: KioskOwnerCap) {
```

Comment:
I kind of hate this API. Why not just have this as a parameter to the function calls that require it? That seems more intuitive.

Reply:
Or alternatively if you break this up into more logical functional units, you could have it as part of the construction of the client instead, which also makes more sense to me.

Reply:
@Jordan-Mysten Yeah we can indeed remove the state from here, and require it on the `ownedKioskTx` or `getOwnerCap` calls instead.